Project-5-Task-1.md

```diff
@@ -1302,9 +1308,7 @@ Zero dimensional spatial region, one dimensional spatial region, two dimensional
 
 ___
```

Comment:
Methodology Review
Group 1 first of all, wow. You have done a lot of work and it is quite impressive. Your axiom knowledge bases are incredibly expansive. Group 2 would like to commend you on your effort and collaboration! I would suggest that you scale down a little bit 😊 but I think you already know that! May I ask how you plan to check your schemas? I am interested to see how you will check the schemes you present. Further, it would be beneficial to see how and why you choose the axioms you did and how this will fit into your schema constructions. I would love to hear about the knowledge graph. I know y’all are working on developing the graph, but I would personally benefit from hearing up on how y’all plan to design the graph. Additionally, will you build one graph or several?
If there is more you would like us to consider please let me know! Overall, Great Work!!

Reply:
Dear @dmcnulty27, 

Thank you so much for your review and your encouraging words! I think we plan on scaling down for sure. I do not think we are able to finish all of these axioms 😅. 

To answer your questions:

(1) We are checking our schemas using a SHACL validator tool which you can find here: https://github.com/TopQuadrant/shacl. Also, we are using SHACL Playground as a secondary supplemental tool. 

(2) On choosing these axioms: We actually did not choose these axioms, they were assigned to us. Professor Beverley wanted us to ensure that our SHACL schema can account for the logical axioms that undergird BFO. The logical axioms have been created primarily by Alan Ruttenberg (according to my limited knowledge). 

(3) On the knowledge graph: We will be building and submitting one graph. The initial Astrea SHACL graph we created is based on the knowledge graph with instance data added, while also being built off of the BFO 2020 ontology without the first order logic axioms added. This project seeks to update that initial SHACL schema with more SHACL shapes that enforce the first order logic axioms. So all the validated SHACL shapes that you see on our Google doc that correspond to a logical axiom will be added to the original Astrea SHACL schema. 

At this link, you can see all of our most updated SHACL enforced FOL work: https://docs.google.com/document/d/1miP3WUXwh3rG3AoBgrDXQrfRwlesV-7XlMhrtjQG2sc/edit#

Thanks again for the encouragement. It's been a hard project and we appreciate the time you spent to review our work. 

Please let me know if you have any questions. 

Best, 
Jaron & Group 1

Reply:
One more Methodology question, coming from a place of curiosity/confusion (rather than critique): There might be a question: if the ontological structure of a project which is supposed to be a BFO structure, but an entity in that project does not fit into any class in BFO, then there should be a Shacl shape checking it.

Reply:
@katiejohnso, thanks for sharing this curiosity! I would be very interested in that particular entity that does not fit in the BFO structure and also, how a SHACL shape would test it! Lol. Thanks for sharing this good thought. It has me thinking now.

Reply:
> One more Methodology question, coming from a place of curiosity/confusion (rather than critique): There might be a question: if the ontological structure of a project which is supposed to be a BFO structure, but an entity in that project does not fit into any class in BFO, then there should be a Shacl shape checking it.

I guess any entity must fit into some class in BFO because BFO is a top-level ontology? Even if we cannot find a subclass which a strange entity falls under, we can always at least subsume it under BFO:entity.